Mathlib/Data/Bool/Basic.lean

```diff
@@ -307,11 +307,23 @@ def toNat (b : Bool) : Nat :=
   cond b 1 0
 #align bool.to_nat Bool.toNat
 
+lemma toNat_le_one (b: Bool) : b.toNat ≤ 1 := by
+  cases b <;> simp only
+
 /-- convert a `ℕ` to a `Bool`, `0 -> false`, everything else -> `true` -/
 def ofNat (n : Nat) : Bool :=
   decide (n ≠ 0)
 #align bool.of_nat Bool.ofNat
 
+@[simp] lemma toNat_true  : toNat true = 1  := rfl
+@[simp] lemma toNat_false : toNat false = 0 := rfl
+
+@[simp] lemma toNat_beq_zero (b) : (Bool.toNat b == 0) = !b := by
+  cases b <;> rfl
+
+@[simp] lemma toNat_bne_zero (b) : (Bool.toNat b != 0) = b := by
+  rw [bne, toNat_beq_zero, not_not]
```

Comment:
Presumably you want versions of these for `== 1` too?

Reply:
Sure, I've added those and made the formatting a bit more compact